src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const modifyedObjects = [];
+  const typeValueAdd = 'addProperties';
+  const typeValueRemove = 'removeProperties';
+  const typeValueClear = 'clear';
+
+  for (const actionObject of actions) {
+    switch (actionObject.type) {
```

Comment:
You can destructure all the needed properties from action (keysToRemove, extraData, type)